src/main/java/org/apache/sling/feature/cpconverter/handlers/slinginitialcontent/SlingInitialContentBundleEntryMetaDataCollector.java

```diff
@@ -106,10 +121,54 @@ Set<SlingInitialContentBundleEntryMetaData> collectFromContextAndWriteTmpFiles()
                 }
             }
         }
-
+        
         return collectedSlingInitialContentBundleEntries;
     }
+    
+    Set<SlingInitialContentBundleEntryMetaData> getDefaultContentXmlEntries() throws UnsupportedEncodingException {
+        
+        createDefaultContentXMLFiles();
+        return defaultContentXmlEntries;
+        
+    }
 
+    private void createDefaultContentXMLFiles() throws UnsupportedEncodingException {
+        
+        for(String repositoryPath: collectedSlingInitialContentBundleEntries.stream()
+                .map(SlingInitialContentBundleEntryMetaData::getRepositoryPath)
+                .collect(Collectors.toList())){
+            addDefaultSlingContentXMLAndParents(repositoryPath);
+        }
+    }
+
+    private void addDefaultSlingContentXMLAndParents(String repositoryPath) throws UnsupportedEncodingException {
+
+        if(StringUtils.countMatches(repositoryPath, '/') < 2){
```

Comment:
I find this opaque - please document the reason.

Reply:
also: could it be that the repositorypath contains a trailing / ? then you would actually continue.....
but similar to roberts finding: what is different with level 1 and 2? that looks brittle....